src/shared/configUtils.ts

```diff
@@ -35,32 +44,36 @@ export class ConfigUtils {
     return this.#globalConfig;
   }
 
-  public static async getOrCreateIdentityToken(): Promise<string> {
-    let token = await this.getIdentityToken();
-    if (!token) {
-      token = CryptoUtils.generateIdentityToken();
-      await this.writeIdentityToken(token);
+  public static async getOrCreateIdentityToken(username: string, tokenService: IdentityTokenService): Promise<string> {
+    let identityData = await this.getIdentityData();
+    if (!identityData) {
+      const token = CryptoUtils.generateIdentityToken();
+      const entityId = await tokenService.saveTokenToServer(token);
+      identityData = {
+        identityToken: token,
+        usernameToServerEntityIdMap: {},
+      };
+      identityData.usernameToServerEntityIdMap[username] = entityId;
+      await this.writeIdentityData(identityData);
+      return token;
+    } else {
+      const entityId = await tokenService.saveTokenToServer(identityData.identityToken);
```

Comment:
We should first check to see if the Entity ID for this user already exists in the configuration, before writing the data to the server. We should only update the app server if there's no entity association locally already.

Reply:
I might have misunderstood your previous [comment](https://github.com/salesforcecli/plugin-lightning-dev/pull/89#discussion_r1671243350). How will we then deal with a situation when the entity, for what ever reason, got deleted on the server?

Reply:
I'd consider getting deleted on the server as a pretty edgy of edge cases—we provide no interface to do that, and users would have to intentionally, manually initiate such action through their own means. In that case, the server should just throw an error, and users can reset their local project state accordingly.

Reply:
Regarding my previous comment, my thinking was more centered around what happens if configuration is lost _locally_, which is more likely because, you know, file system, projects / files move or get deleted, etc. So I wanted a design that could easily recreate the app server association with the local identity, without having hard-to-diagnose sync issues between the local and app server environments.